### ISSUE TITLE
Gradient Clipping for discriminator only

### ISSUE THREAD

**Issue by VasudevGupta7** [PyTorchLightning/pytorch-lightning#2992]
How can I clip the weights for only discriminator while training GAN?

Thanks

**Comment by awaelchli** [comment]
Override the optimizer step in your LightningModule. Example:

```python
class MyModule(LightningModule):

    def optimizer_step(self, current_epoch, batch_idx, optimizer, optimizer_idx, *args, **kwargs):

        if optimizer_idx == 1:  # change accordingly
             # this is the optimizer for discriminator
             # clip your gradients here
             torch.nn.utils.clip_grad_norm_(parameters, ...)

         # do the actual step
         super().optimizer_step(...)
```

https://pytorch-lightning.readthedocs.io/en/latest/lightning-module.html#optimizer-step

**Comment by VasudevGupta7** [comment]
Got it. Thanks for your help.
